cdbsearch.py

```diff
@@ -523,15 +557,19 @@ async def cdbsearch(
         print("  cdb PV len: ", chessdb.cdbPvToLeaf.get(board.epd(), 0), flush=True)
         bestscore, pv = await chessdb.search(board, depth)
         print("  score     : ", bestscore)
-        pvlen = len(pv) - 1 if pv[-1] in ["checkmate", "draw", "invalid"] else len(pv)
+        pvlen = (
+            len(pv) - 1
+            if pv[-1] in ["checkmate", "draw", "EGTB", "invalid"]
+            else len(pv)
+        )
         print("  PV        : ", " ".join(pv[:-1]), end=" ", flush=True)
         if proveMates and pv[-1] == "checkmate" and pvlen:
             if await chessdb.pv_has_proven_mate(board.copy(), pv):
                 pv[-1] = "CHECKMATE" + (
                     f" (#{(pvlen+1)//2})" if bestscore > 0 else f" (#-{pvlen//2})"
                 )
         print(f"{pv[-1]}\n  PV len    :  {pvlen}")
-        print(f"  max ply   :  {len(chessdb.semaphoreTree) - 1}")
+        print(f"  max ply   :  {len(chessdb.semaphoreTree)}")
```

Comment:
I believe the -1 was incorrect. Agreed?

Reply:
Not so sure, I've added it there because we have `self.semaphoreTree[ply]`, in that case `len(semaphreTree)` is ply + 1 ? Am I missing something ?

Reply:
I am going to untangle this PR into the separate pieces. It sort of grew out of control as I was making the TBsearch edits in my branch. There I will explain my reasoning.